src/server/server_family.cc

```diff
@@ -3179,6 +3206,39 @@ void ServerFamily::Module(CmdArgList args, const CommandContext& cmd_cntx) {
   rb->SendLong(20'000);  // we target v2
 }
 
+void ServerFamily::ClientPauseCmd(CmdArgList args, SinkReplyBuilder* builder,
+                                  ConnectionContext* cntx) {
+  CmdArgParser parser(args);
+  auto listeners = GetNonPriviligedListeners();
+
+  auto timeout = parser.Next<uint64_t>();
+  ClientPause pause_state = ClientPause::ALL;
+  if (parser.HasNext()) {
+    pause_state = parser.MapNext("WRITE", ClientPause::WRITE, "ALL", ClientPause::ALL);
+  }
+  if (auto err = parser.Error(); err) {
+    return builder->SendError(err->MakeReply());
+  }
+
+  const auto timeout_ms = timeout * 1ms;
+  auto is_pause_in_progress = [this, end_time = chrono::steady_clock::now() + timeout_ms] {
+    return ServerState::tlocal()->gstate() != GlobalState::SHUTTING_DOWN &&
+           chrono::steady_clock::now() < end_time && client_pause_.load(std::memory_order_relaxed);
+  };
+
+  std::unique_lock lk(client_pause_mu_);
+  if (auto pause_fb_opt =
+          Pause(listeners, cntx->ns, cntx->conn(), pause_state, std::move(is_pause_in_progress));
+      pause_fb_opt) {
+    client_pause_fb_.JoinIfNeeded();
+    client_pause_.store(true, std::memory_order_relaxed);
```

Comment:
only after the previous pasu is finished you set client_pause_ = true , this var name is confusing, it could be that there is no more active pause. lets try to think why we need it and give it better name

Reply:
I agree the name is confusing. We still need this atomic because it provides as cancellation. I will adjust the semantics those and the name of the varialbe. +1

Reply:
Looking for naming suggestions :eyes: